azext_iot/common/_azure.py

```diff
@@ -30,6 +30,11 @@ def parse_iot_device_connection_string(cs):
     return _parse_connection_string(cs, validate, 'Device')
 
 
+def parse_iot_device_module_connection_string(cs):
+    validate = ['HostName', 'DeviceId', 'ModuleId', 'SharedAccessKey']
```

Comment:
Please ensure that this code is not case sensitive.  For example, if someone uses hostname instead of HostName, your code should still work.

Reply:
I think it would work as "parse_iot_hub_connection_string" & "parse_iot_device_connection_string" other methods have same format

Reply:
I suggest writing a test to make sure

Reply:
The user can't submit the connectionstring as it is generated in code with pre-defined template "HostName={};DeviceId={};ModuleId={};{}" mentioned [here](https://github.com/Azure/azure-iot-cli-extension/blob/11d6e4f9a2cb3bd8e69741d5fe9548a60943fb30/azext_iot/operations/hub.py#L801), So it won't be in lowercase.

Reply:
It's irrelevant that the connection string is generated as of right now, as that can change in the future. As Jon suggested, let's just write unit tests to make sure that we accept any combination of lower/upper case inputs.

Reply:
The intent of this PR is to add support of module SAS token generation to the existing `az iot hub generate-sas-token` command. Some test examples for devices are [here](https://github.com/Azure/azure-iot-cli-extension/blob/dev/azext_iot/tests/test_iot_ext_int.py#L344) you can follow them for the module sas token tests and expand on them. Remember to include updated documentation and examples.

For supporting case insensitive connection string input, I recommend submitting a new PR with the concise implementation of that. Currently connection string case sensitivity is strictly enforced.  That being said its a relatively smaller item to support effort wise.

Connection string generation for the user should be strict with respect to specs and documentation as in the casing should remain the same.

Reply:
Fixed case-sensitive issue